### PR TITLE
COP-6

### PR DESCRIPTION
**Pull request by andrew-ironforge**
https://metisoft.atlassian.net/browse/COP-6

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,9 +20,8 @@
   },
   "homepage": "https://github.com/MetisoftSolutions/metisoft-firebase-util#readme",
   "dependencies": {
-    "@types/bluebird": "^3.5.20",
-    "bluebird": "^3.5.1",
-    "firebase-admin": "^5.12.1"
+    "es6-promise": "^4.2.5",
+    "firebase-admin": "^7.0.0"
```
Comment (andrew-ironforge) [inline package.json:24]: Upgraded to support rich notifications. Earlier versions of firebase did not

File: src/index.ts [hunk 2]
```diff
@@ -24,32 +24,53 @@ export interface ConfigurationOptions {
...
+  badge: number;
+  sound?: string;
+}
 
-export function sendPushNotification(userId: string, title: string, body: string): Promise<string|void> {
+export interface ISendPushNotificationOptions {
+  userId: string;
+  title: string;
+  body: string;
+  richOptions?: IRichNotificationOptions;
```
Comment (jbrawnjr) [inline src/index.ts]: `richNotificationOptions`

File: src/index.ts [hunk 3]
```diff
@@ -58,21 +79,45 @@ export function sendPushNotification(userId: string, title: string, body: string
 
     .then(() => {
       let message = {
-            notification: {
-              title,
-              body
-            },
-            token: firebaseToken
-          };
+        notification: {
```
Comment (jbrawnjr) [inline src/index.ts]: Now that we have a lot more logic in this section, let's make a function `__genFirebaseMessage()` that takes `firebaseToken` and `ISendPushNotificationOptions` and returns `admin.messaging.Message`.

File: src/index.ts [hunk 4]
```diff
@@ -88,12 +133,12 @@ export function init(opts: ConfigurationOptions) {
     throw new Error("databaseUrl key in opts required.");
   }
 
-  options = Object.assign(options, opts);
+  __options = Object.assign(__options, opts);
 
-  let serviceAccount = require(options.pathToServiceAccountKey);
+  let serviceAccount = require(opts.pathToServiceAccountKey);
```
Comment (jbrawnjr) [inline src/index.ts]: `opts` -> `__options`

File: src/index.ts [hunk 5]
```diff
@@ -88,12 +133,12 @@ export function init(opts: ConfigurationOptions) {
...
 
   admin.initializeApp({
     credential: admin.credential.cert(serviceAccount),
-    databaseURL: options.databaseUrl
+    databaseURL: opts.databaseUrl
```
Comment (jbrawnjr) [inline src/index.ts]: `opts` -> `__options`

**Review by jbrawnjr — CHANGES_REQUESTED** [review]

**Review by jbrawnjr — APPROVED** [review]

**Pull request merged**